File: samples/java/src/main/java/co/windly/limbosample/presentation/main/MainComponent.java
package co.windly.limbosample.presentation.main;

import co.windly.limbosample.presentation.main.root.RootModule;
import dagger.Subcomponent;
import dagger.android.AndroidInjector;

@Subcomponent(modules = {
  RootModule.class
})
public interface MainComponent extends AndroidInjector<MainActivity> {

  //region Factory

  @Subcomponent.Factory
  interface Factory extends AndroidInjector.Factory<MainActivity> {
    // No-op.
  }

  //endregion
}

File: samples/kotlin/src/main/java/co/windly/limbosample/presentation/main/MainComponent.kt
package co.windly.limbosample.presentation.main

import co.windly.limbosample.presentation.main.home.HomeModule
import co.windly.limbosample.presentation.main.settings.SettingsModule
import dagger.Subcomponent
import dagger.android.AndroidInjector

@Subcomponent(modules = [
  HomeModule::class,
  SettingsModule::class
])
interface MainComponent : AndroidInjector<MainActivity> {

  //region Builder

  @Subcomponent.Factory
  interface Factory : AndroidInjector.Factory<MainActivity>

  //endregion
}
